torch/_inductor/utils.py

```diff
@@ -634,16 +637,31 @@ def has_incompatible_cudagraph_ops(gm):
     return False
 
 
+# Attempt to import AttrsDescriptor from Triton and check for 'ids_of_folded_args'
 try:
-    from triton.compiler.compiler import AttrsDescriptor as instance_descriptor
+    from triton.compiler.compiler import AttrsDescriptor
+    # Check for the presence of 'ids_of_folded_args'
+    has_ids_of_folded_args = 'ids_of_folded_args' in {f.name for f in fields(AttrsDescriptor)}
 except ImportError:
-    # To support older version of triton which does not have AttrsDescriptor
-    # class
-    instance_descriptor = collections.namedtuple(  # type: ignore[no-redef]
-        "instance_descriptor",
-        ["divisible_by_16", "equal_to_1", "ids_of_folded_args", "divisible_by_8"],
-        defaults=[tuple(), tuple(), tuple(), tuple()],
-    )
+    # Fallback if AttrsDescriptor is not available
+    AttrsDescriptor = make_dataclass('AttrsDescriptor',
```

Comment:
Any reason not use normal Python dataclass class syntax here?

Reply:
I think that for its fallback logic the dynamic of `make_dataclass` is preferred. At runtime the effect is the same as it generate the class if not available.